src/metabase/driver/sql/query_processor.clj

```diff
@@ -128,11 +128,30 @@
 (defmulti ->honeysql
   "Return an appropriate HoneySQL form for an object. Dispatches off both driver and either clause name or object class
   making this easy to override in any places needed for a given driver."
-  {:arglists '([driver x])}
+  {:arglists '([driver mbql-expr-or-object])}
   (fn [driver x]
     [(driver/dispatch-on-initialized-driver driver) (mbql.u/dispatch-by-clause-name-or-class x)])
   :hierarchy #'driver/hierarchy)
 
+(defn compiled
```

Comment:
See `throw-double-compilation-error` below for an explanation of why this was added